route-rs-runtime/src/utils/test/harness.rs

```diff
@@ -0,0 +1,66 @@
+use crate::link::{Link, TokioRunnable};
+use crate::utils::test::packet_collectors::ExhaustiveCollector;
+use crossbeam::crossbeam_channel;
+use futures::future::lazy;
+use std::fmt::Debug;
+
+/// The utils::test::harness module should be able to help Link authors abstract away the
+/// complexity of dealing with the Tokio runtime. Tests should be expressed with the
+/// typical "Given, When, Then" structure (https://martinfowler.com/bliki/GivenWhenThen.html).
+
+/// "Given" refers to the state of the world before the behavior under test runs.
+/// The initial context needed to test a link is the incoming packet stream(s).
+
+/// "When" refers to the behavior under test.
+/// This is the Link configuration we're trying to test.
+
+/// "Then" refers to the expected changes to the system due to executing the behavior under test
+/// against the initial context.
+/// This is the state of packet collectors after the input has been exhausted and run through
+/// our Link under test.
+
+/// Since the initial context of "a link's input streams" are coupled to the construction of Links,
+/// let's just expose a function that takes a connected Link, runs it's runnables and collectors
+/// through Tokio, and extracts the output packets into vectors representing egress streams.
+
+// TODO: fn name improvement
```

Comment:
Some other names to consider:
test_link
exercise_link

I don't think run_link is that bad. But if this is a function we only expect to use for tests, then test_link may read more clearly. 
I leave the decision to you, is isn't a big deal